File: dotnet/src/Datastructure/Otterop/Datastructure/StringBuffer.cs
namespace Otterop.Datastructure
{
    public class StringBuffer
    {
        private LinkedList<Otterop.Lang.String> strings;
        public StringBuffer()
        {
            this.strings = new LinkedList<Otterop.Lang.String>();
        }

        public void Add(Otterop.Lang.String s)
        {
            this.strings.AddLast(s);
        }

        public Otterop.Lang.String OOPString()
        {
            Otterop.Lang.OOPIterable<Otterop.Lang.String> strings = this.strings;
            return Otterop.Lang.String.Concat(strings);
        }

    }

}

File: java/datastructure/src/main/java/otterop/datastructure/StringBuffer.java
package otterop.datastructure;

import otterop.lang.OOPIterable;
import otterop.lang.String;

public class StringBuffer {

    private LinkedList<String> strings;

    public StringBuffer() {
        this.strings = new LinkedList<String>();
    }

    public void add(String s) {
        this.strings.addLast(s);
    }

    public String OOPString() {
        OOPIterable<String> strings = this.strings;
        return String.concat(strings);
    }
}
